File: java2csharp/java/Twilio/video-quickstart-android-task-remove-ion-dependency/exampleScreenCapturer/src/main/java/com/twilio/video/examples/screencapturer/ScreenCapturerActivity.java
package com.twilio.video.examples.screencapturer;

import android.content.Context;
import android.content.Intent;
import android.media.projection.MediaProjectionManager;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.util.Log;
import android.view.Menu;
import android.view.MenuInflater;
import android.view.MenuItem;
import android.view.View;
import android.widget.Toast;

import com.twilio.video.LocalMedia;
import com.twilio.video.LocalVideoTrack;
import com.twilio.video.ScreenCapturer;
import com.twilio.video.VideoView;

/**
 * This example demonstrates how to use the screen capturer
 */
public class ScreenCapturerActivity extends AppCompatActivity {
    private static final String TAG = "ScreenCapturer";
    private static final int REQUEST_MEDIA_PROJECTION = 100;

    private LocalMedia localMedia;
    private VideoView localVideoView;
    private LocalVideoTrack screenVideoTrack;
    private ScreenCapturer screenCapturer;
    private MenuItem screenCaptureMenuItem;
    private final ScreenCapturer.Listener screenCapturerListener = new ScreenCapturer.Listener() {
        @Override
        public void onScreenCaptureError(String errorDescription) {
            Log.e(TAG, "Screen capturer error: " + errorDescription);
            stopScreenCapture();
            Toast.makeText(ScreenCapturerActivity.this, R.string.screen_capture_error,
                    Toast.LENGTH_LONG).show();
        }

        @Override
        public void onFirstFrameAvailable() {
            Log.d(TAG, "First frame from screen capturer available");
        }
    };

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_screen_capturer);
        localVideoView = (VideoView) findViewById(R.id.local_video);

        localMedia = LocalMedia.create(this);
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        MenuInflater inflater = getMenuInflater();
        inflater.inflate(R.menu.screen_menu, menu);
        return true;
    }

    @Override
    public boolean onPrepareOptionsMenu(Menu menu) {
        // Grab menu items for updating later
        screenCaptureMenuItem = menu.findItem(R.id.share_screen_menu_item);
        return true;
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        switch (item.getItemId()) {
            case R.id.share_screen_menu_item:
                String shareScreen = getString(R.string.share_screen);

                if (item.getTitle().equals(shareScreen)) {
                    if (screenCapturer == null) {
                        requestScreenCapturePermission();
                    } else {
                        startScreenCapture();
                    }
                } else {
                    stopScreenCapture();
                }

                return true;
            default:
                return super.onOptionsItemSelected(item);
        }
    }

    private void requestScreenCapturePermission() {
        Log.d(TAG, "Requesting permission to capture screen");
        MediaProjectionManager mediaProjectionManager = (MediaProjectionManager)
                getSystemService(Context.MEDIA_PROJECTION_SERVICE);

        // This initiates a prompt dialog for the user to confirm screen projection.
        startActivityForResult(mediaProjectionManager.createScreenCaptureIntent(),
                REQUEST_MEDIA_PROJECTION);
    }

    @Override
    public void onActivityResult(int requestCode, int resultCode, Intent data) {
        if (requestCode == REQUEST_MEDIA_PROJECTION) {
            if (resultCode != AppCompatActivity.RESULT_OK) {
                Toast.makeText(this, R.string.screen_capture_permission_not_granted,
                        Toast.LENGTH_LONG).show();
                return;
            }
            screenCapturer = new ScreenCapturer(this, resultCode, data, screenCapturerListener);
            startScreenCapture();
        }
    }

    private void startScreenCapture() {
        screenVideoTrack = localMedia.addVideoTrack(true, screenCapturer);
        screenCaptureMenuItem.setIcon(R.drawable.ic_stop_screen_share_white_24dp);
        screenCaptureMenuItem.setTitle(R.string.stop_screen_share);

        localVideoView.setVisibility(View.VISIBLE);
        screenVideoTrack.addRenderer(localVideoView);
    }

    private void stopScreenCapture() {
        localVideoView.setVisibility(View.INVISIBLE);
        localMedia.removeVideoTrack(screenVideoTrack);

        screenCaptureMenuItem.setIcon(R.drawable.ic_screen_share_white_24dp);
        screenCaptureMenuItem.setTitle(R.string.share_screen);
        screenVideoTrack.removeRenderer(localVideoView);
    }

    @Override
    protected void onDestroy() {
        if (localMedia != null) {
            if (screenVideoTrack != null) {
                localMedia.removeVideoTrack(screenVideoTrack);
            }
            localMedia.release();
            localMedia = null;
        }
        super.onDestroy();
    }

}

File: java2csharp/csharp/Twilio/voice-and-video-dependencies-w-ion-dependency/Sample.VideoRooms.ScreenCapturer/ScreenCapturerActivity.cs
namespace com.twilio.video.examples.screencapturer
{

	using Context = android.content.Context;
	using Intent = android.content.Intent;
	using MediaProjectionManager = android.media.projection.MediaProjectionManager;
	using Bundle = android.os.Bundle;
	using AppCompatActivity = android.support.v7.app.AppCompatActivity;
	using Log = android.util.Log;
	using Menu = android.view.Menu;
	using MenuInflater = android.view.MenuInflater;
	using MenuItem = android.view.MenuItem;
	using View = android.view.View;
	using Toast = android.widget.Toast;


	/// <summary>
	/// This example demonstrates how to use the screen capturer
	/// </summary>
	public class ScreenCapturerActivity : AppCompatActivity
	{
		private const string TAG = "ScreenCapturer";
		private const int REQUEST_MEDIA_PROJECTION = 100;

		private LocalMedia localMedia;
		private VideoView localVideoView;
		private LocalVideoTrack screenVideoTrack;
		private ScreenCapturer screenCapturer;
		private MenuItem screenCaptureMenuItem;
		private readonly ScreenCapturer.Listener screenCapturerListener = new ListenerAnonymousInnerClassHelper();

		private class ListenerAnonymousInnerClassHelper : ScreenCapturer.Listener
		{
			public ListenerAnonymousInnerClassHelper()
			{
			}

			public override void onScreenCaptureError(string errorDescription)
			{
				Log.e(TAG, "Screen capturer error: " + errorDescription);
				outerInstance.stopScreenCapture();
				Toast.makeText(outerInstance, R.@string.screen_capture_error, Toast.LENGTH_LONG).show();
			}

			public override void onFirstFrameAvailable()
			{
				Log.d(TAG, "First frame from screen capturer available");
			}
		}

		public override void onCreate(Bundle savedInstanceState)
		{
			base.onCreate(savedInstanceState);
			ContentView = R.layout.activity_screen_capturer;
			localVideoView = (VideoView) findViewById(R.id.local_video);

			localMedia = LocalMedia.create(this);
		}

		public override bool onCreateOptionsMenu(Menu menu)
		{
			MenuInflater inflater = MenuInflater;
			inflater.inflate(R.menu.screen_menu, menu);
			return true;
		}

		public override bool onPrepareOptionsMenu(Menu menu)
		{
			// Grab menu items for updating later
			screenCaptureMenuItem = menu.findItem(R.id.share_screen_menu_item);
			return true;
		}

		public override bool onOptionsItemSelected(MenuItem item)
		{
			switch (item.ItemId)
			{
				case R.id.share_screen_menu_item:
					string shareScreen = getString(R.@string.share_screen);

					if (item.Title.Equals(shareScreen))
					{
						if (screenCapturer == null)
						{
							requestScreenCapturePermission();
						}
						else
						{
							startScreenCapture();
						}
					}
					else
					{
						stopScreenCapture();
					}

					return true;
				default:
					return base.onOptionsItemSelected(item);
			}
		}

		private void requestScreenCapturePermission()
		{
			Log.d(TAG, "Requesting permission to capture screen");
			MediaProjectionManager mediaProjectionManager = (MediaProjectionManager) getSystemService(Context.MEDIA_PROJECTION_SERVICE);

			// This initiates a prompt dialog for the user to confirm screen projection.
			startActivityForResult(mediaProjectionManager.createScreenCaptureIntent(), REQUEST_MEDIA_PROJECTION);
		}

		public override void onActivityResult(int requestCode, int resultCode, Intent data)
		{
			if (requestCode == REQUEST_MEDIA_PROJECTION)
			{
				if (resultCode != AppCompatActivity.RESULT_OK)
				{
					Toast.makeText(this, R.@string.screen_capture_permission_not_granted, Toast.LENGTH_LONG).show();
					return;
				}
				screenCapturer = new ScreenCapturer(this, resultCode, data, screenCapturerListener);
				startScreenCapture();
			}
		}

		private void startScreenCapture()
		{
			screenVideoTrack = localMedia.addVideoTrack(true, screenCapturer);
			screenCaptureMenuItem.Icon = R.drawable.ic_stop_screen_share_white_24dp;
			screenCaptureMenuItem.Title = R.@string.stop_screen_share;

			localVideoView.Visibility = View.VISIBLE;
			screenVideoTrack.addRenderer(localVideoView);
		}

		private void stopScreenCapture()
		{
			localVideoView.Visibility = View.INVISIBLE;
			localMedia.removeVideoTrack(screenVideoTrack);

			screenCaptureMenuItem.Icon = R.drawable.ic_screen_share_white_24dp;
			screenCaptureMenuItem.Title = R.@string.share_screen;
			screenVideoTrack.removeRenderer(localVideoView);
		}

		protected internal override void onDestroy()
		{
			if (localMedia != null)
			{
				if (screenVideoTrack != null)
				{
					localMedia.removeVideoTrack(screenVideoTrack);
				}
				localMedia.release();
				localMedia = null;
			}
			base.onDestroy();
		}

	}

}